src/jiraIssueStatusUpdates/issueTransitioner.gs

```diff
@@ -0,0 +1,121 @@
+// Node required code block
+const Request = require('../jiraApi.gs');
+const getIssue = require('../jiraCommon.gs').getIssue;
+const IssueTransitionConfiguration = require('./issueTransitionConfiguration.gs');
+// End of Node required code block
+
+/**
+ * Gets the list of transitions valid for the specified issue in its current status
+ * @param {string} issueKey the id of the issue
+ */
+function getPossibleTransitions(issueKey) {
+  var response = {
+    'data': [],
+    'status': -1,
+    'errorMessage': ''
+  };
+
+  var ok = function (resp, httpResp, status) {
+    if (resp && resp.transitions) {
+      response = {
+        'data': resp,
+        'status': status,
+        'errorMessage': resp.hasOwnProperty('warningMessages') ? resp.warningMessages : 'No results found.'
+      };
+    } else {
+      // Something funky is up with the JSON response.
+      debug.error("Failed to retrieve transition issue data for ID [" + issueKey + "]! resp:%s; httpResp:%s; status:%s", resp, httpResp, status);
+    }
+  };
+
+  var error = function (resp, httpResp, status) {
+    debug.error('[%s] %s - %s', status, resp, httpResp);
+
+    var msgs = resp.hasOwnProperty('errorMessages') ? resp.errorMessages : [];
+    msgs = msgs.concat((resp.hasOwnProperty('warningMessages') ? resp.warningMessages : []));
+
+    response.status = status;
+    response.errorMessage = "Could not fetch issues transition states: " + msgs.join("\n");
+  };
+
+  var request = new Request();
+  request.call('issueTransitions', { issueIdOrKey: issueKey })
+    .withSuccessHandler(ok)
+    .withFailureHandler(error);
+
+  return response;
+}
+
+/**
+ * Makes a request to the JIRA API to apply the transition to an issue
+ * @param {string} issueKey issue key 
+ * @param {integer} transitionId the id of the transition to move it to the desired state
+ */
+function makeTransition(issueKey, transitionId) {
+  var request = new Request();
+  request.call('issueTransitionUpdate', { issueIdOrKey: issueKey, "transition": { "id": transitionId } });
+  var resp = request.getResponse();
+  returnData = { success: false, errorMessage: null };
+  returnData.success = (resp.statusCode == 204);
+  if (!returnData.success) {
```

Comment:
I started using the pattern callbacks of ok and error because I thought request.call was asynchronous. (the usual pattern for JS in Node and the browser). However it's not in GAS
I prefer this for readability. 
Not against changing it though.